### DISCUSSION TITLE
how to use Azure Service Fabric StateManger to save and transfer job/schedule states

### DISCUSSION THREAD

**Discussion by alanzeng6181** [quartznet/quartznet#473] [General]
I am using Quartz.net inside a ServiceFabric StatefulService, which has a StateManager (IReliableStateManager) that manages states. I implemented a custom class called AzureQuartzJobStore that basically mimics RAMJobStore, except states such as HashSet of JobKey, SortedSet of TriggerWrapper are saved to StateManager. 

However, in the case of failover, StateManager would transfer/sync the states from one server to another, but how do I activate the Triggers for the jobs in the new server?  Or is the way I am doing the wrong approach?

Thanks!

**Comment by lahma** [comment]
I'm afraid that I'm not familiar with ServiceFabric to give any advice. Generally new job store sounds like a way, and of course if you can utilize Azure SQL services then AdoJobStore should fit your need.

**Comment by alanzeng6181** [comment]
@lahma Thank you! I will look into Azure SQL.
